geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/Conversions.scala

```diff
@@ -201,8 +201,11 @@ object RichSimpleFeatureType {
   // in general we store everything as strings so that it's easy to pass to accumulo iterators
   implicit class RichSimpleFeatureType(val sft: SimpleFeatureType) extends AnyVal {
 
-    def getGeomField: String = sft.getGeometryDescriptor.getLocalName
-    def getGeomIndex: Int = sft.indexOf(sft.getGeometryDescriptor.getLocalName)
+    def getGeomField: String = {
```

Comment:
Ideally, yes, but in practice I don't think so... almost everywhere we use it, we're in code that is excluded up front by the table.supports mechanism. So basically we would be doing getGeomField.getOrElse(throw new Exception). The places we aren't protected I added checks.
Yeah, it might lead to some subtle bugs down the road, but I think sfts without geoms is kind of a fringe case anyway...